src/net/sourceforge/kolmafia/session/ResponseTextParser.java

```diff
@@ -146,44 +146,75 @@ public static boolean hasResult(final String location) {
       queryString = location.substring(queryStringBegin + 1);
     }
 
+    // *** This is an unfinished and unused feature.
     if (!PageRegistry.isGameAction(path, queryString)) {
       return false;
     }
 
-    if (location.startsWith("lchat.php")
-        || location.startsWith("mchat.php")
-        || location.startsWith("newchatmessages.php")
-        || location.startsWith("submitnewchat.php")) {
-      return false;
-    }
-
-    if (location.startsWith("showplayer.php")) {
-      // We want to register new items from Equipment and Familiar
-      return true;
-    }
-
-    if (location.startsWith("displaycollection.php")) {
-      // We want to register new items in display cases
-      return true;
+    switch (path) {
+      case "login.php", "logout.php" -> {
+        return false;
+      }
+      case "api.php", "game.php", "doc.php" -> {
+        return false;
+      }
+      case "topmenu.php", "awesomemenu.php" -> {
+        return false;
+      }
+      case "actionbar.php" -> {
+        return false;
+      }
+      case "lchat.php", "mchat.php", "newchatmessages.php", "submitnewchat.php" -> {
+        // Nothing from chat.
+        return false;
+      }
+      case "desc_effect.php",
+          "desc_familiar.php",
+          "desc_guardian.php",
+          "desc_item.php",
+          "desc_monster.php",
```

Comment:
`desc_monster.php` doesn't seem to exist on KoL. There is a reference to it in Mafia, but I don't understand what it is supposed to do.

Reply:
Yeah, that is a synthetic page that we use in the Encyclopedia for the "Monsters" tab.
Every other tab will display the page returned by KoL itself.

It is trickery in the MonsterDescriptionFrame:

```
  private void refreshMonster() {
    MonsterData stats = null;

    // Kludge for a single monster...
    if (this.currentMonster.getName().equals("Ed the Undying")) {
      String name = "Ed the Undying (" + (variant + 1) + ")";
      stats = MonsterDatabase.findMonster(name);
    }

    String path = "desc_monster.php?whichmonster=" + this.currentMonster.getId();
    GenericRequest request = new GenericRequest(path);
    request.responseText = this.currentMonster.craftDescription(this.variant, stats);
    this.refresh(request);
  }
```

Notice that it makes a GenericRequest with a pseudo-url and sets the responseText to hand-crafted HTML.
It never actually runs that request. It just displays the responsetext.

I can remove desc_monster.php, since we never submit that and look for a response to parse.